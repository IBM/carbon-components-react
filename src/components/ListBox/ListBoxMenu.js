import React from 'react';
import PropTypes from 'prop-types';

const ListBoxMenu = ({ children, ...rest }) => {
  return (
    <div className="bx--list-box__menu" {...rest}>
      {children}
    </div>
  );
};

ListBoxMenu.propTypes = {
  children: PropTypes.node,
};

export default ListBoxMenu;
