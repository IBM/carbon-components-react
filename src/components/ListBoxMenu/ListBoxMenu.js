import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Icon from '../Icon';

const handleOnClick = event => {
  // Prevent default interaction here so that the event bubbles up to
  // the `onClick` handler supplied by `getItemProps`
  event.preventDefault();
};

const ListBoxMenu = ({
  items,
  selectedItem,
  highlightedIndex,
  itemToString,
  getItemProps,
}) => (
  <div className="bx--list-box__menu">
    {items.map((item, i) => {
      const itemProps = getItemProps({ item, index: i });
      return (
        <div
          key={itemProps.id}
          className={cx({
            'bx--list-box__menu-item': true,
            'bx--list-box__menu-item--highlighted': highlightedIndex === i,
            'bx--list-box__menu-item--active': selectedItem.indexOf(item) !== -1,
          })}
          {...itemProps}>
          <input
            id={itemProps.id}
            name={itemToString(item)}
            className="bx--checkbox"
            type="checkbox"
            checked={selectedItem.indexOf(item) !== -1}
            readOnly={true}
            tabIndex="-1"
            onClick={handleOnClick}
          />
          <label htmlFor={itemProps.id} className="bx--checkbox-label">
            <span className="bx--checkbox-appearance">
              <Icon className="bx--checkbox-checkmark" name="checkmark" />
            </span>
            {itemToString(item)}
          </label>
        </div>
      );
    })}
  </div>
);

ListBoxMenu.propTypes = {
  items: PropTypes.array.isRequired,
  selectedItem: PropTypes.array.isRequired,
  highlightedIndex: PropTypes.number,
  itemToString: PropTypes.func.isRequired,
  getItemProps: PropTypes.func.isRequired,
};

export default ListBoxMenu;
