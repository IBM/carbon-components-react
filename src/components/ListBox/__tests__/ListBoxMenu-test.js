import React from 'react';
import { mount } from 'enzyme';
import ListBoxMenu from '../ListBoxMenu';

describe('ListBoxMenu', () => {
  it('should render', () => {
    const wrapper = mount(
      <ListBoxMenu>
        <span>Hello</span>
      </ListBoxMenu>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
