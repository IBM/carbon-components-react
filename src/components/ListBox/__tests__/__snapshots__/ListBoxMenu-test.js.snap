// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxMenu should render 1`] = `
<ListBoxMenu>
  <div
    className="bx--list-box__menu"
  >
    <span>
      Hello
    </span>
  </div>
</ListBoxMenu>
`;
